src/x11.rs

```diff
@@ -74,19 +105,96 @@ impl X11Impl {
         let visual = (lib.XDefaultVisual)(display_handle.display as *mut Display, screen);
         let depth = (lib.XDefaultDepth)(display_handle.display as *mut Display, screen);
 
+        // See if we can load the XShm extension.
+        let xshm = XShm::open()
+            .ok()
+            .filter(|shm| (shm.XShmQueryExtension)(display_handle.display as *mut Display) == 0);
+
         Ok(Self {
             window_handle,
             display_handle,
-            lib,
+            xlib: lib,
+            xshm: xshm.map(|xshm| ShmExtension { xshm, shmseg: None }),
             gc,
             visual,
             depth,
         })
     }
 
     pub(crate) unsafe fn set_buffer(&mut self, buffer: &[u32], width: u16, height: u16) {
+        if let Err(e) = self.shm_set(buffer, width, height) {
+            eprintln!("XShm not available, falling back to XImage: {e}");
```

Comment:
We should probably use the `log` crate instead of println

Reply:
I would prefer using the `tracing` crate if possible.

Reply:
Looks like `winit` uses `log`. Might make sense to use the same thing?

This should probably be a debug message if it's completely expected for some X servers or a remote connection, and doesn't indicate any kind of error.